hooks/emitter/decoder.go

```diff
@@ -661,6 +670,39 @@ func DecodeDescription(des stakingtypes.Description) common.JsDict {
 	}
 }
 
+func DecodeFeedsMsgSubmitSignalPrices(msg *feedstypes.MsgSubmitSignalPrices, detail common.JsDict) {
+	detail["validator"] = msg.GetValidator()
+	detail["timestamp"] = msg.GetTimestamp()
+	detail["signal_prices"] = msg.GetSignalPrices()
+}
+
+func DecodeFeedsMsgVote(msg *feedstypes.MsgVote, detail common.JsDict) {
+	detail["voter"] = msg.GetVoter()
+	detail["signals"] = msg.GetSignals()
+}
+
+func DecodeFeedsMsgUpdateReferenceSourceConfig(msg *feedstypes.MsgUpdateReferenceSourceConfig, detail common.JsDict) {
+	rsc := msg.GetReferenceSourceConfig()
+	detail["registry_ipfs_hash"] = rsc.RegistryIPFSHash
+	detail["registry_version"] = rsc.RegistryVersion
+}
+
+func DecodeFeedsMsgUpdateParams(msg *feedstypes.MsgUpdateParams, detail common.JsDict) {
+	params := msg.GetParams()
+	detail["authority"] = msg.GetAuthority()
+	detail["admin"] = params.GetAdmin()
+	detail["allowable_block_time_discrepancy"] = params.GetAllowableBlockTimeDiscrepancy()
+	detail["grace_period"] = params.GetGracePeriod()
+	detail["min_interval"] = params.GetMinInterval()
+	detail["max_interval"] = params.GetMaxInterval()
+	detail["power_step_threshold"] = params.GetPowerStepThreshold()
+	detail["max_current_feeds"] = params.GetMaxCurrentFeeds()
+	detail["cooldown_time"] = params.GetCooldownTime()
+	detail["min_deviation_basis_point"] = params.GetMinDeviationBasisPoint()
+	detail["max_deviation_basis_point"] = params.GetMaxDeviationBasisPoint()
+	detail["current_feeds_update_interval"] = params.GetCurrentFeedsUpdateInterval()
+}
```

Comment:
can't we just detail["params"] = msg.GetParams()?   not sure if possible please help verify.

Reply:
ok